### ISSUE TITLE
dmd build error

### ISSUE THREAD

**Issue by paumor** [NixOS/nixpkgs#85804]
**Describe the bug**
Wanted to build tilix on nixos-unstable, but dmd throws an error

**To Reproduce**
try to add tilix to confiugration or run:
```
nix-shell -p dmd
```

**Expected behavior**
dmd is build without errors

**Additional context**
issue occurd while updateting my systemconfiguration

**Notify Mainters**
Thomas Mader <thomas.mader@gmail.com>

**Metadata**
```
dmd/complex.d(57): Deprecation: opMul is deprecated.  Use opBinary(string op)(...) if (op == "*") instead.
         dmd/complex.d(57): Deprecation: opMul_r is deprecated.  Use opBinaryRight(string op)(...) if (op == "*") instead.
         dmd/complex.d(62): Deprecation: opMul is deprecated.  Use opBinary(string op)(...) if (op == "*") instead.
         dmd/complex.d(62): Deprecation: opMul_r is deprecated.  Use opBinaryRight(string op)(...) if (op == "*") instead.
         dmd/complex.d(67): Deprecation: opDiv is deprecated.  Use opBinary(string op)(...) if (op == "/") instead.
         dmd/constfold.d(98): Deprecation: opNeg is deprecated.  Use opUnary(string op)() if (op == "-") instead.
         dmd/constfold.d(214): Deprecation: opAdd is deprecated.  Use opBinary(string op)(...) if (op == "+") instead.
         dmd/constfold.d(317): Deprecation: opSub is deprecated.  Use opBinary(string op)(...) if (op == "-") instead.
         dmd/constfold.d(369): Deprecation: opMul is deprecated.  Use opBinary(string op)(...) if (op == "*") instead.
         dmd/constfold.d(369): Deprecation: opMul_r is deprecated.  Use opBinaryRight(string op)(...) if (op == "*") instead.
         dmd/constfold.d(411): Deprecation: opDiv is deprecated.  Use opBinary(string op)(...) if (op == "/") instead.
         make[1]: *** [posix.mak:472: ../generated/linux/release/64/dmd] Error 1
         make[1]: Leaving directory '/build/dmd/src'
         make: *** [posix.mak:8: all] Error 2
         builder for '/nix/store/pmn03rw4b76akr0kkl39i3rgwgyb4bf2-dmd-2.085.1.drv' failed with exit code 2
         error: build of '/nix/store/pmn03rw4b76akr0kkl39i3rgwgyb4bf2-dmd-2.085.1.drv' failed
```

**Comment by veprbl** [comment]
Is this not fixed by #85768 ?

**Comment by worldofpeace** [comment]
@veprbl It builds for me
